### DISCUSSION TITLE
Records Stack Overflow - Compiler Bug?

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#4000] [General] [answered]
I think this is a compiler bug.

Run this code:
```
using System;

namespace RecordTests {
    class Program {
        static void Main(string[] args) {
            var G = Glance.None;

            Console.WriteLine(G.ToString());

        }
    }

    public record Glance {
        public static Glance None { get; private set; } = new Glance();

        public Glance() {

        }
     
    }

}
```

This results in code similar to the following:
```
        [NullableContext(1)]
        protected virtual bool PrintMembers(StringBuilder builder)
        {
            builder.Append("None");
            builder.Append(" = ");
            builder.Append(Glance.None);
            return true;
        }

        public override string ToString()
        {
            StringBuilder stringBuilder = new StringBuilder();
            stringBuilder.Append("Glance");
            stringBuilder.Append(" { ");
            if (this.PrintMembers(stringBuilder))
            {
                stringBuilder.Append(" ");
            }
            stringBuilder.Append("}");
            return stringBuilder.ToString();
        }
```

Which causes a stack overflow because it recursively tries to print the static ```None``` member.

This looks like a bug to me.

**Comment by TonyValenti** [comment, marked as answer]
My suggestion is that record's ```ToString``` should not print out static members.

**Reply by PathogenDavid** [reply]
This was already done in https://github.com/dotnet/csharplang/pull/3919 and https://github.com/dotnet/roslyn/pull/47868, it's just not in the latest previews. ([Example of the new behavior on SharpLab](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxAgrgOyQExAagB8ABAJgEYBYAKBoCUACBBgXgczgHcAKASgG4axcgE5uCAHQAVAPYBlGAgCWmAOZ8BNGgjgBjGQhwM6NAN40GlhsQDM18gDYGKmAwAarBuQAM3wdStrOxcGAE1PUl9/AF8aIA=))
